apps/frontend/src/lib/psi/index.ts

```diff
@@ -22,20 +23,81 @@ export const generatePSIKeyPair = async () => {
   };
 };
 
-export const generateBitVectorFromLannaData = (
-  data: LannaData
-): Uint32Array => {
-  const bitVector = new Uint32Array(1000).fill(0);
-
-  // 0-7 reserved for desired connections
-  bitVector[0] = data.desiredConnections.getHealthy ? 1 : 0;
-  bitVector[1] = data.desiredConnections.cowork ? 1 : 0;
-  bitVector[2] = data.desiredConnections.enjoyMeals ? 1 : 0;
-  bitVector[3] = data.desiredConnections.learnFrontierTopics ? 1 : 0;
-  bitVector[4] = data.desiredConnections.findCollaborators ? 1 : 0;
-  bitVector[5] = data.desiredConnections.goExploring ? 1 : 0;
-  bitVector[6] = data.desiredConnections.party ? 1 : 0;
-  bitVector[7] = data.desiredConnections.doMentalWorkouts ? 1 : 0;
-
-  return bitVector;
+const PSI_BIT_VECTOR_SIZE = 10000;
+const COMMON_CONNECTIONS_BIT_VECTOR_SIZE = 9000;
+
+export const generateBitVectorFromUserData = (
+  userData: UserData,
+  connections: Record<string, Connection>
+): { bitVector: Uint32Array; indexMapping: Record<number, string[]> } => {
+  const bitVector = new Uint32Array(PSI_BIT_VECTOR_SIZE).fill(0);
+  const bitIndexToUsername: Record<number, string[]> = {};
+
+  // 0-8999 reserved for common connections
```

Comment:
So you're leaving 9000 places open to ensure there's enough space for connections. How do all these empty indices affect the computation? Is it efficient enough that it doesn't matter?

Reply:
yeah, this is just based on previous experimentation, i think index size 100,000 is where begins to be a ux hit